plugins/arStorageServicePlugin/modules/arStorageService/actions/extractFileAction.class.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class arStorageServiceExtractFileAction extends sfAction
+{
+  public function execute($request)
+  {
+    $view = sfView::NONE;
+
+    $this->resource = $this->getRoute()->resource;
+
+    // Check that object exists and that it is not the root
+    if (!isset($this->resource))
+    {
+      $this->forward404();
+    }
+
+    try
+    {
+      $status = $this->extractFile($request);
+    }
+    catch (QubitApi404Exception $e)
+    {
+      $this->response->setStatusCode(404, $e->getMessage());
+      throw $e;
+    }
+    catch (QubitApiNotAuthorizedException $e)
+    {
+      $this->response->setStatusCode(401, $e->getMessage());
+      throw $e;
+    }
+    catch (QubitApiForbiddenException $e)
+    {
+      $this->response->setStatusCode(403, $e->getMessage());
+      throw $e;
+    }
+    catch (QubitApiBadRequestException $e)
+    {
+      $this->response->setStatusCode(400, $e->getMessage());
+      throw $e;
+    }
+    catch (Exception $e)
+    {
+      $this->response->setStatusCode(500, $e->getMessage());
+
```

Comment:
Nitpick: either this blank line should be removed, or a blank line should be added before the preceding "throws" for consistency.  I don't know if our style guide suggests a blank line before a throw, or not?

Reply:
I think you are correct. I have removed any blank lines before a 'throw' statement.